spring-kafka/src/main/java/org/springframework/kafka/streams/RecoveringDeserializationExceptionHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.streams;
+
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.streams.errors.DeserializationExceptionHandler;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.kafka.listener.ConsumerRecordRecoverer;
+
+/**
+ * A {@link DeserializationExceptionHandler} that calls a {@link ConsumerRecordRecoverer}.
+ * and continues.
+ *
+ * @author Gary Russell
+ * @since 2.3
+ *
+ */
+public class RecoveringDeserializationExceptionHandler implements DeserializationExceptionHandler {
+
+	/**
+	 * Property name for the application context id.
+	 */
+	public static final String KSTREAM_DESERIALIZATION_CONTEXT_ID = "spring.deserialization.context.id";
+
+	/**
+	 * Property name for the bean name of the recoverer.
+	 */
+	public static final String KSTREAM_DESERIALIZATION_RECOVERER = "spring.deserialization.recoverer";
+
+	private static final Log LOGGER = LogFactory.getLog(RecoveringDeserializationExceptionHandler.class);
+
+	private ConsumerRecordRecoverer recoverer;
```

Comment:
`final` and `@Nullable` and also initialized as a `null` in the ctor.

Reply:
? It's assigned in `configure()`.

Reply:
Ah! That's all about `toString()` on objects in that case.

And there is no way to pass object at all:
```
public DeserializationExceptionHandler defaultDeserializationExceptionHandler() {
        return getConfiguredInstance(DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, DeserializationExceptionHandler.class);
    }
```

Then OK: let's live with what you have so far. Until Apache Kafka will come up with the setters on the `KafkaStreams`...

Reply:
Actually, the problem is ours - `KafkaStreamsConfiguration.asProperties()` - we are doing conversion there. However, they are using the `Properties` as a general Map, with any object type. It's a bit of an abuse but I think we can change that method to just `putAll()`.